### PR TITLE
Support for Rustls

### LINKED ISSUE #43: Add Rustls support
Now that the code has been updated to `tokio` 0.2 (thank you again for that!), it should be possible with not too much work to support `rustls` to add an alternative to `native-tls`.

### PR DESCRIPTION
**Pull request by Geobert**
fixes #43 

I know you won't integrate it before stabilizing 0.7 but I make it available for review for whenever you can :)

PR done on behalf of Isode Ltd

### CODE REVIEW

File: src/conn.rs [hunk 1]
```diff
@@ -14,36 +14,60 @@ use crate::RequestId;
...
+#[cfg(any(feature = "tls-native", feature = "tls-rustls"))]
 use tokio::sync::oneshot;
 use tokio::time;
-#[cfg(feature = "tls")]
+#[cfg(all(feature = "tls-native", not(feature = "tls-rustls")))]
 use tokio_native_tls::{TlsConnector as TokioTlsConnector, TlsStream};
+#[cfg(all(feature = "tls-rustls", not(feature = "tls-native")))]
+use tokio_rustls::{client::TlsStream, TlsConnector as TokioTlsConnector};
+#[cfg(not(any(feature = "tls-native", feature = "tls-rustls")))]
+compile_error!(r#"Either feature "tls-native" or "tls-rustlsls" must be enabled for this crate"#);
```
Comment (inejge) [inline src/conn.rs:40]: It must be possible to compile the crate without _any_ TLS support, so this test is superfluous.

File: src/conn.rs [hunk 2]
```diff
@@ -363,6 +386,63 @@ impl LdapConnAsync {
...
+            Some(connector) => connector,
+            None => LdapConnAsync::create_connector(&settings),
+        };
+        TokioTlsConnector::from(connector)
+            .connect(hostname, stream)
+            .await
+            .map_err(LdapError::from)
+    }
+
+    #[cfg(not(feature = "tls-native"))]
```
Comment (inejge) [inline src/conn.rs:405]: This should be `cfg(feature = "tls-rustls")` to cover the case with no TLS at all.

**Comment by inejge** [comment]
Thanks for the PR! I gave it a brief look and noticed just one issue: the changes wouldn't allow compiling the crate with no TLS support. I've left the comments in the code. When testing, try compiling with `--no-default-features`.

File: src/conn.rs (hunk 2 above)
Comment (Geobert) [reply src/conn.rs:405]: It's what I did before, but than the all-feature flag won't work :-/

Comment (Geobert) [reply src/conn.rs:405]: Don't mind this comment, I tested it and it works :)

**Comment by Geobert** [comment]
Thanks for the merge! I though it would not be included before 0.7.1? Nice surprise :D

**Comment by inejge** [comment]
Okay, I updated this a bit (`native-tls` can be chosen by specifying `tls` as the feature name, as before) and merged, since I'll publish another alpha because of other changes -- see #48, you'll be pleasantly surprised, I hope 🙂

**Pull request merged**